lib/absinthe/schema/notation.ex

```diff
@@ -1683,12 +1682,16 @@ defmodule Absinthe.Schema.Notation do
     [{Absinthe.Middleware.MapGet, identifier}]
   end
 
-  def __ensure_middleware__(middleware, field, _object) do
-    if Absinthe.Type.meta(field, :absinthe_telemetry) do
-      [{Absinthe.Middleware.Telemetry, []} | middleware]
-    else
-      middleware
-    end
+  # Don't install Telemetry middleware for Introspection fields
+  @introspection [Absinthe.Phase.Schema.Introspection, Absinthe.Type.BuiltIns.Introspection]
+  def __ensure_middleware__(middleware, %{definition: definition}, _object)
+      when definition in @introspection do
+    middleware
+  end
+
+  # Install Telemetry middleware
+  def __ensure_middleware__(middleware, _field, _object) do
+    [{Absinthe.Middleware.Telemetry, []} | middleware]
```

Comment:
This installs it _literally_ every field right? This will have non trivial overhead on MapGet fields.

Reply:
No I don't believe so, my understanding is that the `__ensure_middleware__` function heads above install `Absinthe.Middleware.MapGet` alone for the default middleware case...

Reply:
@benwilson512 ping